pysindy/feature_library/base.py

```diff
@@ -506,6 +506,117 @@ def calc_trajectory(self, diff_method, x, t):
         return self.libraries[0].calc_trajectory(diff_method, x, t)
 
 
+class TrimmedLibrary(BaseFeatureLibrary):
```

Comment:
When we made the `EnsembleOptimizer`, we lost some functionality in notebook 13, which I noticed when I went through the example notebooks now. It's also useful for implicit SINDy to be able to mask a feature library term if you want to use it in place of `x_dot`, as in `SINDyPI`. I added it here just to get the example notebook 13 running again, but it is useful to be able to construct a library then remove specific terms from it. Maybe it would be better to make another pull request, but I only had a week where I could focus on pysindy, so I snuck it in here. I have to turn to other work now.

Reply:
The quick fix, if you don't have time to focus on pysindy, is to either let example 13 be broken or, remove `TrimmedLibrary` from here and put it in notebook 13.  The enterprise-level fix is to remove it from this PR, add an issue for what was broken in 13 and why a `TrimmedLibrary` is a useful solution, and handle it separately with a new test.

Broadly speaking, I don't think we don't need to maintain the notebooks unless they implement a change for fast testing in #203 (see README.rst).  It's too much of a burden otherwise, since they take a huge amount of time and memory to run.  When someone runs into an error, they can make an issue and we can build a test, a fix, and if someone has the time, implement the maintainability fixes from the #203.

The broken notebooks serve as an example of what pysindy can do, even if they don't show the modern syntax.

Reply:
I agree that reproduciblility and supporting recent graduates is important.  But "How to keep one piece of software perfectly reproducible while working on another version" is a solved problem: add a requirements for each of the notebooks, and use a separate environment to satisfy those dependencies.  At the bare minimum, just put the version of pysindy that the experiments require at the top of the file: 1.7.4 or 1.7.5.  The alternative is never developing a new major version and never using a new major version of numpy, scikit-learn, matplotlib, etc, requiring future students and postdocs to maintain not just pysindy, but someone else's research project.

I didn't think you or Alan were that attached to the notebooks all being able to run, since when I started contributing a couple years ago, some of them were already broken.  Nevertheless, I do care about supporting your and Alan's research longevity, and that includes across major releases.  So I built #203 to make it easier for you and Alan to protect the functionality of your notebooks across major versions (in particular, see [the readme part](https://github.com/dynamicslab/pysindy#contributing-examples)).

There's probably a discussion to be had about examples vs full research papers as notebooks in the repo, as well as whether we should always should maintain backwards compatibility of a notebook across major versions even if they could have fast tests.  When does reading a bit of updated documentation or troubleshooting an error become negligible compared to reading the notebook & paper in the first place?

Reply:
I'm trying to be respectful, but discussions like this are not a good use of resources. I can spend time explaining decisions to you that are important, but that's not really my job.  

You can open the notebooks, read the papers, understand what they are trying to do, and figure out why I made the changes if you want. You'll see that the features in the TrimmedLibrary are exactly those that were lost in notebook 13, but getting the notebook running is not the only reason to include it. I explained the use case with SINDyPI and ensemble SINDy already above, and I wanted to be able to test those things on research questions. 

I know you want me to make another pull request for it and you will weigh in on its value, but I don't want to waste a bunch of time writing about trivial changes and discussing them. If you disagree with my decisions, then I will hear out your rationale, but the costs of contributing here are outweighing the benefits for me. 

This repository is not an enterprise endeavor, and good scientific practices (particularly, not wasting time and computing resources) are at least as important as enterprise coding practices here in my opinion. We can have a difference of values, but we shouldn't let that become a barrier to collaboration.